### PR TITLE
North West 4 - Morteza Khojasteh - JavaScript Core 1 - Week 3 (Mandatory)

### PR DESCRIPTION
**Pull request by Morteza-Khojasteh**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

**Review by humailhasankhan — COMMENTED** [review]
Looking at your code is so pleasing to the eyes. Your oxygenLevel exercise seems to be much shorter and concise than mine. I like it. I was just wondering how did you tackle the password validator. That one was very difficult for me and took me a lot of time to figure out how to make sure the password does not repeat. I don't see your condition for that. Otherwise, Great job.

**Comment by Morteza-Khojasteh** [comment]
> Looking at your code is so pleasing to the eyes. Your oxygenLevel exercise seems to be much shorter and concise than mine. I like it. I was just wondering how did you tackle the password validator. That one was very difficult for me and took me a lot of time to figure out how to make sure the password does not repeat. I don't see your condition for that. Otherwise, Great job.

Thanks for comment Humail

File: 2-mandatory/1-create-functions.js [hunk 1]
```diff
@@ -24,7 +29,11 @@ Write a function that:
 - Removes any forward slashes (/) in the strings.
 - Makes the strings all lowercase.
 */
-function tidyUpString() {
+function tidyUpString(arr) {
+  const tidyArr = arr.map((element) => {
+    return element.trim().replace("/", "").toLowerCase();
```
Comment (lannem) [inline 2-mandatory/1-create-functions.js:34]: nice 👍

File: 2-mandatory/1-create-functions.js [hunk 2]
```diff
@@ -33,7 +42,8 @@ Write a function that:
 - Returns a new array containing the same elements, but without the element at the passed index.
 */
 
-function remove() {
+function remove(arr, index) {
+  return arr.slice(0, index).concat(arr.slice(index + 1));
```
Comment (lannem) [inline 2-mandatory/1-create-functions.js:46]: nice! You could also use `.splice()` which lets you pass in the start and end indexes of the elements you want to remove

File: 2-mandatory/1-create-functions.js [hunk 3]
```diff
@@ -44,9 +54,18 @@ Write a function that:
...
 
-function formatPercentage() {
+function formatPercentage(arr) {
+  return arr.map((number) => {
+    if (number >= 100) {
+      return "100%";
+    } else {
+      return Math.round(number * 100) / 100 + "%";
+    }
+  });
```
Comment (lannem) [inline 2-mandatory/1-create-functions.js:64]: this is nice and clean well done!

File: 2-mandatory/4-space-colonies.js [hunk 4]
```diff
@@ -15,7 +15,13 @@
   
 */
 
-function getSettlers() {}
+function getSettlers(arr) {
+  return arr.filter(
+    (element) => (element.includes("family")) && (element.startsWith("A")));
```
Comment (lannem) [inline 2-mandatory/4-space-colonies.js:20]: nice

**Review by lannem — COMMENTED** [review]
great work morteza well done:)

File: 2-mandatory/1-create-functions.js (hunk 1 above)
Comment (Morteza-Khojasteh) [reply 2-mandatory/1-create-functions.js:34]: Thanks

File: 2-mandatory/1-create-functions.js (hunk 2 above)
Comment (Morteza-Khojasteh) [reply 2-mandatory/1-create-functions.js:46]: Yes of course

File: 2-mandatory/1-create-functions.js (hunk 3 above)
Comment (Morteza-Khojasteh) [reply 2-mandatory/1-create-functions.js:64]: Thanks

**Pull request closed without merging**